nntrainer/tensor/manager.cpp

```diff
@@ -416,28 +422,45 @@ std::vector<Weight *> Manager::requestWeights(
       // var_exec_order.push_back(TensorPool::PERSIST_END_ORDER);
     }
 
-    Tensor *var = nullptr, *grad = nullptr;
+    Tensor *var = nullptr, *grad = nullptr, *var_m = nullptr;
     bool is_dependent = !shared_names.empty();
+    TensorDim dim_a = dim;
     if (is_dependent) {
       /// shared_name is used and the orignal name is discarded
       const auto &shared_name = shared_names.at(i);
       /** case when shared names are given */
-      var = weight_pool.requestOrExtend(shared_name, dim, var_exec_order,
-                                        var_ls, t_initializer);
+      if (act_type == TensorDim::DataType::NONE) {
```

Comment:
can't this be replaced as follows?
```suggestion
      if (act_type == dim.getDataType()) {
```

Reply:
it used to check "not specified" state. so, it's necessary.